test/components/tableTests.ts

```diff
@@ -294,77 +118,346 @@ describe("Tables", () => {
       table = new Plottable.Components.Table([[c1]]);
       c1.detach();
-      assert.deepEqual((<any> table)._rows, [[null]], "calling detach() on the Component removed it from the Table");
+      assert.isNull(table.componentAt(0, 0), "calling detach() on the Component removes it from the Table");
       assert.isNull(c1.parent(), "Component disconnected from the Table");
     });
   });
 
-  describe("checking Table contents", () => {
-    it("has()", () => {
-      let c0 = new Plottable.Component();
-      let table = new Plottable.Components.Table([[c0]]);
-      assert.isTrue(table.has(c0), "correctly checks that Component is in the Table");
-      table.remove(c0);
-      assert.isFalse(table.has(c0), "correctly checks that Component is no longer in the Table");
-      table.add(c0, 1, 1);
-      assert.isTrue(table.has(c0), "correctly checks that Component is in the Table again");
+  describe("requesting space", () => {
+    it("is fixed-width if all columns contain only fixed-width Components, non-fixed otherwise", () => {
+      const fixedNullsTable = new Plottable.Components.Table([
+        [new Mocks.FixedSizeComponent(), null],
+        [null, new Mocks.FixedSizeComponent()]
+      ]);
+      assert.isTrue(fixedNullsTable.fixedWidth(), "width is fixed if all columns contain fixed-width Components or null");
+
+      const someUnfixedTable = new Plottable.Components.Table([
+        [new Mocks.FixedSizeComponent(), new Mocks.FixedSizeComponent()],
+        [new Plottable.Component(), null]
+      ]);
+      assert.isFalse(someUnfixedTable.fixedWidth(), "width is not fixed if any column contains a non-fixed-width Component");
     });
 
-    it("can retrieve the Component at a given row, column index", () => {
-      let c00 = new Plottable.Component();
-      let c01 = new Plottable.Component();
-      let c10 = new Plottable.Component();
-      let c11 = new Plottable.Component();
-      let table = new Plottable.Components.Table([
-        [c00, c01],
-        [c10, c11]
+    it("is fixed-height if all rows contain only fixed-height Components, non-fixed otherwise", () => {
+      const fixedNullsTable = new Plottable.Components.Table([
+        [new Mocks.FixedSizeComponent(), null],
+        [null, new Mocks.FixedSizeComponent()]
       ]);
+      assert.isTrue(fixedNullsTable.fixedHeight(), "height is fixed if all columns contain fixed-height Components or null");
 
-      assert.strictEqual(table.componentAt(0, 0), c00, "retrieves the Component at [0, 0]");
-      assert.strictEqual(table.componentAt(0, 1), c01, "retrieves the Component at [0, 1]");
-      assert.strictEqual(table.componentAt(1, 0), c10, "retrieves the Component at [1, 0]");
-      assert.strictEqual(table.componentAt(1, 1), c11, "retrieves the Component at [1, 1]");
+      const someUnfixedTable = new Plottable.Components.Table([
+        [new Mocks.FixedSizeComponent(), new Plottable.Component()],
+        [new Mocks.FixedSizeComponent(), null]
```

Comment:
It proves that having any one row non-fixed is sufficient to make the `Table` non-fixed, even if other rows are fixed.